worker/manager/src/mirrors_qa_manager/worker.py

```diff
@@ -0,0 +1,309 @@
+# pyright: strict, reportMissingTypeStubs=false, reportUnknownMemberType=false, reportOptionalSubscript=false, reportUnknownVariableType=false, reportUnknownArgumentType=false
+import datetime
+import shutil
+import signal
+import sys
+import time
+from pathlib import Path
+from typing import Any
+from urllib.parse import urlencode
+
+from docker.models.containers import Container
+from docker.types import Mount
+
+from mirrors_qa_manager import logger
+from mirrors_qa_manager.backend import authenticate
+from mirrors_qa_manager.backend import query_api as query_backend_api
+from mirrors_qa_manager.cryptography import load_private_key_from_path
+from mirrors_qa_manager.docker import (
+    exec_command,
+    get_docker_client,
+    query_host_mounts,
+    remove_container,
+    run_container,
+)
+from mirrors_qa_manager.settings import Settings
+
+
+class WorkerManager:
+    """Manager responsible for creating tasks"""
+
+    def __init__(self, worker_id: str) -> None:
+
+        self.worker_id = worker_id
+        self.base_dir = Settings.WORKDIR_FPATH
+
+        # Create the directory where this worker will write their data
+        self.instance_dir = self.base_dir / worker_id
+        self.instance_dir.mkdir(exist_ok=True)
+
+        self.private_key = load_private_key_from_path(Settings.PRIVATE_KEY_FPATH)
+        self.docker = get_docker_client()
+
+        self.host_mounts = query_host_mounts(
+            self.docker,
+            [Settings.WORKDIR_FPATH, Settings.WIREGUARD_KERNEL_MODULES_FPATH],
+        )
+        self.host_workdir = self.host_mounts[Settings.WORKDIR_FPATH]
+
+        self.wg_container_name = f"worker-{self.worker_id}-wireguard"
+        self.task_container_name = f"worker-{self.worker_id}-task"
+        # Create the root directory for binding with the worker's wireguard container
+        self.wg_root_dir = self.instance_dir / "wireguard"
+        self.wg_root_dir.mkdir(exist_ok=True)
+
+        # Create the directory where the configuration file for the interface
+        # will be found.
+        # https://github.com/linuxserver/docker-wireguard/blob/master/root/etc/s6-overlay/s6-rc.d/init-wireguard-confs/run#L5
+        self.wg_confs_dir = self.wg_root_dir / "wg_confs"
+        self.wg_confs_dir.mkdir(exist_ok=True)
+        # name of the wireguard interface
+        self.wg_interface = "wg0"
+        self.wg_healthcheck_cmd = [
+            "curl",
+            "--interface",
+            self.wg_interface,
+            "-s",
+            "https://am.i.mullvad.net/json",
+        ]
+        # commands for bringing down/up the interface whenever a new configuration
+        # file is added
+        self.wg_down_cmd = ["wg-quick", "down", self.wg_interface]
+        self.wg_up_cmd = ["wg-quick", "up", self.wg_interface]
+
+        # Check in with the backend API
+        self.auth_credentials = authenticate(self.private_key, self.worker_id)
```

Comment:
this could fail so this should be caught and handled properly. A failing to start manager needs to indicate problem clearly